src/animated-agent.js

```diff
@@ -0,0 +1,426 @@
+    // time spent collecting garbage and object creation. The cancel member is
+    // used to stop an existing animation to remove race conditions from
+    // multiple animations running on the same Animated.
+    this.animations = {};
+    // Dictionary of Animated property animatedKey to style applied by
+    // AnimatedAgent. Used internally to reapply the style when its removed
+    // when updating `this.rects` in cases like the window resizing.
+    this.styles = {};
+    // Dictionary of Animated property animatedKey to style replaced on an
+    // element. Used internally to remove applied style when updating
+    // `this.rects`.
+    this.replacedStyles = {};
+    // Pool of AnimatedTimer objects that can be reused. This helps reduce
+    // object creation and garbage collection.
+    this.timerPool = [];
+    // Pool of AnimatedCallbackOptions objects that can be reused. This helps
+    // reduce object creation and garbage collection.
+    this.optionsPool = [];
+    // Is this rendered in a browser or on the server as a string.
+    this.clientRender = typeof window !== 'undefined';
+    // Hold a promise that resolves soon. Used by soon() to group a set of calls
+    // as soon as possible, not immediately but also not on the next javascript
+    // tick caused by something like setTimeout. As its a promise it uses the
+    // Promise spec's detail that a resolved promise calls its callbacks at the
+    // end of the current javascript tick before the javascript engine hands off
+    // to the browser to do rendering.
+    this._soon = null;
+  }
+
+  getChildContext() {
+    return {animationAgent: this};
+  }
+
+  componentDidMount() {
+    if (this.clientRender) {
+      window.addEventListener('resize', this.handleResize);
+      window.addEventListener('orientationchange', this.handleResize);
+
+      // Determine the agent's starting rectangle.
+      AnimatedRect.getBoundingClientRect(findDOMNode(this), this.rect);
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.clientRender) {
+      // Determine the agent's updated rectangle.
+      const lastRect = this.rect.clone();
+      AnimatedRect.getBoundingClientRect(findDOMNode(this), this.rect);
+      if (!lastRect.equal(this.rect)) {
+        this.resize(this.soon());
+      }
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.clientRender) {
+      window.removeEventListener('resize', this.handleResize);
+      window.removeEventListener('orientationchange', this.handleResize);
+    }
+  }
+
+  removeAnimatedStyle(animated, animatedEl) {
+    // Return an animated element to the style an animation replaced. This
+    // should return the element to how it was before the animation was played.
+    // This is used at times to return the element to the non animated state to
+    // query the DOM's layout. After use, the style is returned to that of the
+    // any current animation so a user never sees the change.
```

Comment:
Yes that is something I want to do. A lot of the current docs are "internal". But would probably have lots of content cross over.